apcd-cms/src/apps/submission_form/apcd_database.py

```diff
@@ -0,0 +1,362 @@
+            host=APCD_DB['host'],
+            dbname=APCD_DB['database'],
+            user=APCD_DB['user'],
+            password=APCD_DB['password'],
+            port=APCD_DB['port'],
+            sslmode='require'
+        )
+        query = """SELECT
+                registration_contacts.registration_contact_id,
+                registration_contacts.registration_id,
+                registration_contacts.notify_flag,
+                registration_contacts.contact_role,
+                registration_contacts.contact_name,
+                registration_contacts.contact_phone,
+                registration_contacts.contact_email
+                FROM registration_contacts"""
+        cur = conn.cursor()
+        cur.execute(query)
+        return cur.fetchall()
+
+    except Exception as error:
+        logger.error(error)
+
+    finally:
+        if cur is not None:
+            cur.close()
+        if conn is not None:
+            conn.close()
+
+
+def create_registration_contact(form, reg_id, iteration):
+    values = (
+        reg_id,
+        True if 'contact_notifications' in form else False,
+        form['contact_type'],
+        form['contact_name'],
+        re.sub("[^0-9]", "", form['contact_phone']),
+        form['contact_email']
+    )
+    if iteration > 1:
+        if not acceptable_contact(form, iteration):
+            return
+        values = (
+            reg_id,
+            True if 'contact_notifications_{}'.format(iteration) in form else False,
+            form['contact_type_{}'.format(iteration)],
+            form['contact_name_{}'.format(iteration)],
+            re.sub("[^0-9]", "", form['contact_phone_{}'.format(iteration)]),
+            form['contact_email_{}'.format(iteration)]
+        )
+
+    operation = """INSERT INTO registration_contacts(
+        registration_id,
+        notify_flag,
+        contact_role,
+        contact_name,
+        contact_phone,
+        contact_email
+    ) VALUES (%s,%s,%s,%s,%s,%s)"""
+
+    cur = None
+    conn = None
+    try:
+        conn = psycopg2.connect(
+            host=APCD_DB['host'],
+            dbname=APCD_DB['database'],
+            user=APCD_DB['user'],
+            password=APCD_DB['password'],
+            port=APCD_DB['port'],
+            sslmode='require'
+        )
+        cur = conn.cursor()
+        cur.execute(operation, values)
+        conn.commit()
+
+    except Exception as error:
+        logger.error(error)
+
+    finally:
+        if cur is not None:
+            cur.close()
+        if conn is not None:
+            conn.close()
+
+
+def acceptable_entity(form, iteration):
+    entity_keys = [
+        'total_claims_value_{}'.format(iteration),
+        'claims_encounters_volume_{}'.format(iteration),
+        'license_number_{}'.format(iteration),
+        'naic_company_code_{}'.format(iteration),
+        'total_covered_lives_{}'.format(iteration),
+        'entity_name_{}'.format(iteration),
+        'fein_{}'.format(iteration)
```

Comment:
**Question.** The `fein` field (no integer) is captured, right? (And similar such fields.) (Maybe not in this section of code, but generally.)

Reply:
Yes they are, these things here are just to check that the additional groups are valid (complete)